File: 2022/Round 1/B.java
import java.util.*;
import java.io.*;

public class B {
    public static void main(String[] args) throws IOException {
        // TODO: Change the input file name for submission
        String filename = "watering_well_chapter_2_validation_input.txt";

        BufferedReader br = new BufferedReader(new FileReader(filename));
        PrintWriter pw = new PrintWriter(new FileWriter("output.txt"));
        int T = Integer.parseInt(br.readLine());
        final long NUM = 1000000007;

        for (int t = 1; t <= T; t++) {
            int N = Integer.parseInt(br.readLine());
            long[] A = new long[N], B = new long[N];

            for (int i = 0; i < N; i++) {
                StringTokenizer st = new StringTokenizer(br.readLine());
                A[i] = Long.parseLong(st.nextToken());
                B[i] = Long.parseLong(st.nextToken());
            }
            int Q = Integer.parseInt(br.readLine());
            long[] X = new long[Q], Y = new long[Q];

            for (int j = 0; j < Q; j++) {
                StringTokenizer st = new StringTokenizer(br.readLine());
                X[j] = Long.parseLong(st.nextToken());
                Y[j] = Long.parseLong(st.nextToken());
            }
            long ans = 0;
            long x_sum = 0, y_sum = 0;

            for (int j = 0; j < Q; j++) {
                ans += (N * (X[j] * X[j] + Y[j] * Y[j]) % NUM) % NUM;
                ans %= NUM;
                x_sum += X[j];
                y_sum += Y[j];
                x_sum %= NUM;
                y_sum %= NUM;
            }
            for (int i = 0; i < N; i++) {
                ans += (Q * (A[i] * A[i] + B[i] * B[i]) % NUM) % NUM;
                ans -= (2 * (A[i] * x_sum + B[i] * y_sum) % NUM) % NUM;
                ans %= NUM;
            }
            if (ans < 0) {
                ans += NUM;
            }

            pw.println(String.format("Case #%d: %d", t, ans));
        }
        pw.flush();
        pw.close();
        br.close();
    }
}

File: 2022/Round 1/B.py
def main():
    filename = "watering_well_chapter_2_input.txt"
    f = open(filename, "r")
    out = open("output.txt", "w")

    T = int(f.readline())
    NUM = 1000000007

    for t in range(1, T + 1):
        N = int(f.readline())
        A, B = [], []
        for i in range(N):
            a, b = (int(s) for s in f.readline().split(" "))
            A.append(a)
            B.append(b)

        Q = int(f.readline())
        X, Y = [], []
        for j in range(Q):
            x, y = (int(s) for s in f.readline().split(" "))
            X.append(x)
            Y.append(y)

        ans = 0
        x_sum, y_sum = sum(X), sum(Y)

        for j in range(Q):
            ans += N * ((X[j] * X[j] + Y[j] * Y[j]) % NUM)
            ans %= NUM

        for i in range(N):
            ans += Q * ((A[i] * A[i] + B[i] * B[i]) % NUM)
            ans -= 2 * (A[i] * x_sum + B[i] * y_sum) % NUM
            ans %= NUM

        if ans < 0:
            ans += NUM

        out.write(f"Case #{t}: {ans}\n")


if __name__ == "__main__":
    main()
